core/src/main/java/com/spotify/missinglink/ClassLoader.java

```diff
@@ -101,10 +102,31 @@ public DeclaredClass load() throws IOException {
         }
         if (insn instanceof MethodInsnNode) {
           final MethodInsnNode minsn = (MethodInsnNode) insn;
+          boolean isStatic;
+          boolean isVirtual;
+          switch (minsn.getOpcode()) {
+            case Opcodes.INVOKEVIRTUAL:
+            case Opcodes.INVOKEINTERFACE:
+              isVirtual = true;
+              isStatic = false;
+              break;
+            case Opcodes.INVOKESPECIAL:
+              isVirtual = false;
+              isStatic = false;
+              break;
+            case Opcodes.INVOKESTATIC:
+              isVirtual = false;
+              isStatic = true;
+              break;
```

Comment:
I think this switch block could be simplified - that you could remove lines for setting the flags to false since that is the default value of a boolean (and you break after each assignment)

          switch (minsn.getOpcode()) {
            case Opcodes.INVOKEVIRTUAL:
            case Opcodes.INVOKEINTERFACE:
              isVirtual = true;
              break;
            case Opcodes.INVOKESTATIC:
              isStatic = true;
              break;
            default:
               ...